### ISSUE TITLE
💀 ERROR: "GODOT": wget: missing URL

### ISSUE THREAD

**Issue by zen0bit** [ivan-hc/AM#231]
I tried to install godot, but after installation godot can't be started.

Also tried with debug, here is output:
```
am -i --debug godot

 ############################################################################
 ##                                                                        ##
 ##                  START OF ALL INSTALLATION PROCESSES                   ##
 ##                                                                        ##
 ############################################################################

 💀 ERROR: "--debug" does NOT exist in the database, see "am -l"
 ____________________________________________________________________________

 ◆ "GODOT": starting installation script
wget: missing URL
Usage: wget [OPTION]... [URL]...

Try `wget --help' for more options.
unzip:  cannot find or open ./*.zip, ./*.zip.zip or ./*.zip.ZIP.

No zipfiles found.
rm: cannot remove './*.zip': No such file or directory
mv: cannot stat './tmp/*': No such file or directory
chmod: cannot access '/opt/godot/godot': No such file or directory
./godot: 64: ./godot: not found
./godot: 65: ./godot: not found
./godot: 66: ./godot: not found
mv: cannot stat 'squashfs-root/*.desktop': No such file or directory
mv: cannot stat 'squashfs-root/share/applications/*.desktop': No such file or directory
mv: cannot stat 'squashfs-root/usr/local/share/applications/*.desktop': No such file or directory
rm: cannot remove './godot.desktop': No such file or directory
./godot: 71: ./godot: not found
mv: cannot stat 'squashfs-root/usr/local/share/applications/*.desktop': No such file or directory
rm: cannot remove './godot.desktop': No such file or directory
./godot: 75: ./godot: not found
mv: cannot stat 'squashfs-root/share/applications/*.desktop': No such file or directory
cat: ./godot.desktop: No such file or directory
sed: can't read ./godot.desktop: No such file or directory
sed: can't read ./godot.desktop: No such file or directory
sed: can't read ./godot.desktop: No such file or directory
sed: can't read ./godot.desktop: No such file or directory
cat: ./godot.desktop: No such file or directory
sed: can't read ./godot.desktop: No such file or directory
./godot: 87: ./godot: not found
./godot: 88: ./godot: not found
./godot: 89: ./godot: not found
./godot: 90: ./godot: not found
./godot: 91: ./godot: not found
./godot: 92: ./godot: not found
mv: cannot stat './godot.desktop': No such file or directory
                                                   "GODOT" INSTALLED (1 MB OF DISK SPACE)
 ____________________________________________________________________________
 ############################################################################

                  END OF ALL INSTALLATION PROCESSES

             The following new programs have been installed:

 ◆ godot

 ############################################################################
```

**Comment by zen0bit** [comment]
`◆  godot         |  unknown   |  44    KiB`

am thinking that godot is properly installed!

possible issues:
1. wrong godot link
2. Maybe add check if any part of installation process failed..
3. --debug don't show: 💀 ERROR: "--debug" does NOT exist in the database, see "am -l" (I think its working properly except showing this unneeded message)

**Comment by ivan-hc** [comment]
Thank you @zen0bit 

1. the script was wrong due to some fixes in bulk of old scripts, about one year ago
2. this is difficult, all depends on the installation script, and not all apps have the same structure, I can't do much
3. options --debug and --force-latest were blank, I've added a command `return 0` to skipt their "turn"

**Comment by zen0bit** [comment]
> About point 2, all installation scripts should be wrote again.

why?

> Their current state allows them to be used independently

I like that, leave it, if is the only reason to remade (don't make them dependent on am)

Maybe check if end part is succesfull?
Like creating desktop file in path for program.

is desktop file created? y/n
is pointing to real executable file? y/n

Something like that should be enough?

**Comment by ivan-hc** [comment]
No, not enough, unfortunately. .desktop files may be pre-created in some cases.

AM also handles other types of programs, even "godot" is a binary executable file, not an AppImage (as it incorrectly appeared in the old script).

If AM had only handled AppImage, it would have had no problems.

If we then want to include the fact that not all AppImage scripts are the same, then this remains a basic problem.

Other package managers may have a reference size estimate, in the database... but not AM.

Alternatively, I should include a command to check for executable binaries either in $PATH or in /opt/$APP... but that might include scripts, and besides "AM-updater" and "remove" there are actual scripts and just like programs... and other programs are in subfolders of /opt/$APP.

It's a mess.

Therefore, the best thing to do would be to rewrite all installation scripts, including this information to verify.

**Comment by ivan-hc** [comment]
@zen0bit see https://github.com/ivan-hc/AM/commit/d8c3a436649aafaa07d36ad84346f300c18cb9ff

I updated the module to prevent applications with "version" files (therefore, almost all) from being installed if the file is empty.

The "version" file is generated immediately after the download is done.

The "version" file contains the value of the "$version" variable, which is typically a URL or (in some cases) a number.

If the "$version" value is empty, the "version" file is still created, but empty. In this case you have not downloaded any application or absolutely nothing from the internet. In that case, the app is deleted.

**Comment by ivan-hc** [comment]
@zen0bit 

> I updated the module to prevent applications with "version" files (therefore, almost all)

Now we can say that about all the apps have this version file, also a dummy one.

The ones without it are the various portable Firefox/Thunar editions, Tor Browser, Waterfox, scripts and some very small archives (platform-tools and ventoy)
